### PR TITLE
Focus text field when focusing spinner

### PR DESCRIPTION
**Pull request by highfieldjames**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/web/issues/102624

When a `TextField` `Spinner` is clicked on, the button is the document's active element (i.e. the button is focused). When hitting the enter key, if the `TextField` is in a form, the form should be submitted, but instead nothing happens. This differs from the default behaviour of an `<input type="number">` where the input is focused after clicking a spinner.

See this codesandbox for a replication of the issue: https://codesandbox.io/s/cranky-rgb-ktnqml?file=/App.tsx

Note that despite the button being focused previously, hitting enter had no effect, so this PR does not break or change existing functionality, only fixes broken behaviour.

### WHAT is this pull request doing?

This PR changes the TextField component so the `input` is focused when the stepper is clicked.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

[Tophat on Storybook](https://5d559397bae39100201eedc1-wekmtcxvyp.chromatic.com/?path=/story/all-components-textfield--with-form-submit)

1. Click a stepper to increase the text field numerical value
2. Hit enter
3. The form should submit, and the text field should reset to '0'. Previously nothing happened.

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/warm-points-punch.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Focus on TextField on Spinner click
```
Comment (chloerice) [inline .changeset/warm-points-punch.md]:
```suggestion
Updated `TextField` of `type` `number` to focus when a `Spinner` button is clicked
```

File: polaris-react/src/components/TextField/TextField.stories.tsx [hunk 2]
```diff
@@ -829,3 +830,27 @@ export function All() {
...
+  return (
+    <Form
+      onSubmit={(event) => {
+        event.preventDefault();
+        setTextFieldValue('0');
+      }}
+    >
+      <FormLayout>
+        <TextField
+          label="Store name"
```
Comment (chloerice) [inline polaris-react/src/components/TextField/TextField.stories.tsx]: Can this example be a made to be a (simplified) realistic use case, maybe a purchase quantity of a line item of a draft order, or some other case?

Comment (highfieldjames) [reply polaris-react/src/components/TextField/TextField.stories.tsx]: Thanks for the suggestion! I've updated this with an example of updating the on hand inventory quantity, showing the total to make the form submission more obvious.

**Review by chloerice — DISMISSED** [review]
This makes sense, thank you for contributing @highfieldjames 🎉

**Review by chloerice — APPROVED** [review]

**Review by brandon-yetman — APPROVED** [review]
✅  nice improvement!

**Pull request merged**